src/requests/consumption.js

```diff
@@ -751,11 +774,11 @@ const processTargetingCriteria = (targetingData) => {
  * @param {Object} programData - Program template data
  * @returns {Object} - Response contains formatted template
  */
-const formatProgramTemplate = (programData) => {
+const formatProgramTemplate = async (programData) => {
 	try {
 		let programDocument = {}
 		if (programData?.targeting_criteria) {
-			const targeting = processTargetingCriteria(programData?.targeting_criteria)
+			const targeting = await processTargetingCriteria(programData?.targeting_criteria)
 			programDocument.scope = targeting.scope
```

Comment:
@adithyadinesh0412  where we are checking processTargetingCriteria returns success true ? If any chance it throws error roles and metaInfo wont be there